shared/pubsub.js

```diff
@@ -198,7 +198,11 @@ const defaultSocketEventHandlers = {
       default: break
     }
     if (this.shouldReconnect && this.options.reconnectOnDisconnection) {
-      this.scheduleConnectionAttempt()
+      if (this.failedConnectionAttempts <= this.options.maxRetries) {
+        this.scheduleConnectionAttempt()
+      } else {
+        this.destroy()
+      }
```

Comment:
So, does this mean that there's a situation where the user must refresh the page in order for the connection to re-establish itself?

What if, for example, their internet goes down, but it comes back up later. Does your code handle instantly re-connecting when the `online` event is fired?

Reply:
> does this mean that there's a situation where the user must refresh the page in order for the connection to re-establish itself?

Yes that's the point. Such situations would also include unrecoverable errors, or a malicious user being autobanned, if I understand correctly.

> Does your code handle instantly re-connecting when the online event is fired?

It is supposed to do so if possible, unless there is a good reason not to.

Reply:
> It is supposed to do so if possible, unless there is a good reason not to.

Forgive me, this is a complicated PR and I haven't wrapped my mind over ever piece of it yet, so my question for you is: will it be able to still do that if you've called `this.destroy()` here?

Reply:
Once the pubsub instance has been destroyed, it should not reconnect again.

However we can still listen to network events on the `window` object, then maybe use the `BannerGeneral` component to notify the user once they are back online and tell them to reload the page to resync the data.

This might sound inconvenient, but for now it is likely necessary anyway in order to get contract updates and pubsub messages that were missed while being offline.
Well, unless we make resynchronization work on the `online` event like it currently works on page reload, but this is likely out of the scope of this PR, as this issue likely already existed with Primus.

Also, this `destroy()` call will happen after `maxRetries` reconnection attempts, which already amounts to ~6 minutes with `maxRetries: 10`, and to ~15 minutes with `maxRetries: 20`.

This is a lot of time compared to other socket apps like Slack or Discord which completely hide their UI after roughly one minute of offline time, likely to protect their users from the stale data.

Reply:
Also, this PR is not really supposed to add new functionality: all these reconnection capabilities were already available with Primus, albeit untested in the context of the app.

Reply:
OK, in that case can you add a comment here to make sure we don't forget to deal with this, since at some point in the future we probably will? I.e. ideally the app shouldn't require a refresh if there isn't a connection for 15 minutes.

So something like:

```
this.destroy() // TODO: handle app reconnecting when connection returns
```

Reply:
If the app shouldn't require a refresh after only 15 minutes then we can increase that number to whatever appropriate value, even 1 hour for example. However after calling `destroy()` the web socket client won't check for connectivity anymore, which is why this should only be called after `maxRetries` attempts or when an irrecoverable error has occured, since it basically means to give up the reconnection process.

Reply:
Ah, I see. OK that makes sense.